README.md

```diff
@@ -542,6 +542,36 @@ showCard(Pair('X', '♠'));
 // ! TypeError: ‘showCard’ expected a value of type (Pair Rank Suit) as its first argument; received Pair("X", "♠")
 ```
 
+#### `EnumType`
+
+`EnumType` is used to construct [enumerated types][5].
+
+```haskell
+EnumType :: [Any] -> Type
+```
+
+For example:
+
+```javascript
+//    TimeUnit :: Type
+const TimeUnit = $.EnumType(['milliseconds', 'seconds']);
+
+const def = $.create($.env.concat([TimeUnit, $.ValidDate, $.ValidNumber]));
+
+//    convertTo :: TimeUnit -> ValidDate -> ValidNumber
+const convertTo =
+def('convertTo',
+    {},
+    [TimeUnit, $.ValidDate, $.ValidNumber],
+    (unit, date) => date.valueOf() * (unit === 'milliseconds' ? 1000 : 1));
+    
+convertTo('milliseconds', new Date(1));
+// => 1000
+
+convertTo('days', new Date(1));
+// => ! TypeError: ‘convertTo’ expected a value of type ("milliseconds" | "seconds") as its first argument; received "days"
```

Comment:
The `=>` is intended to be read _evaluates to_. The `!` is intended to be read _throws_. We want `!` here but not `=>`.